### ISSUE TITLE
codegen for enum with empty string

### ISSUE THREAD

**Issue by yuyupopo** [koxudaxi/datamodel-code-generator#380]
**Is your feature request related to a problem? Please describe.**

### Environment

`datamodel-code-generator==0.9.3`

### Scenario

Wanted to generate pydantic models from docker engine API. 

`datamodel-codegen --url "https://converter.swagger.io/api/convert?url=https://docs.docker.com/engine/api/v1.30.yaml"`

Note that docker engine api uses openapi 2.0 spec. Therefore, I used `converter.swagger.io` to translate it into 3.0.0.

### Error message

```Python
Traceback (most recent call last):
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/__main__.py", line 424, in main
    strict_types=config.strict_types,
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/__init__.py", line 311, in generate
    results = parser.parse()
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/base.py", line 386, in parse
    self.parse_raw()
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/openapi.py", line 34, in parse_raw
    [*path_parts, '#/components', 'schemas', obj_name],
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/jsonschema.py", line 997, in parse_raw_obj
    self.parse_obj(name, JsonSchemaObject.parse_obj(raw), path)
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1002, in parse_obj
    self.parse_object(name, obj, path)
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/jsonschema.py", line 624, in parse_object
    fields=self.parse_object_fields(obj, path),
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/jsonschema.py", line 571, in parse_object_fields
    field_type = self.parse_enum(field_name, field, [*path, field_name])
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/parser/jsonschema.py", line 820, in parse_enum
    name=self.model_resolver.get_valid_name(field_name),
  File "<my_path>/python3.7/site-packages/datamodel_code_generator/reference.py", line 362, in get_valid_name
    if name[0] == '#':
IndexError: string index out of range
```

### Cause

The docker engine openapi spec defines a schema with an Enum that contains an empty string:

```yaml
    RestartPolicy:
      type: object
      properties:
        Name:
          type: string
          description: ...
          enum:
          - ""   # here
          - always
          - unless-stopped
          - on-failure
```

It seems that [`get_valid_name`](https://github.com/koxudaxi/datamodel-code-generator/blob/2e7545f728775d4a80fe5f8f4be8e461a0d3dae7/datamodel_code_generator/reference.py#L357
) cannot handle empty strings. 

**Describe the solution you'd like**

I've tested the following workaround by changing the function `get_valid_name` as below.

```python
def get_valid_name(self, name: str, camel: bool = False) -> str:
    if not name:    # +
        return "_"     # +
    if name.isidentifier():
        return name
    if name[0] == '#':
        name = name[1:]   # note that original error is caused here. 
    ...
```

The resulting enum is 

```python
class Name(Enum):
    """
        - Empty string means not to restart
    - `always` Always restart
    - `unless-stopped` Restart always except when the user has manually stopped the container
    - `on-failure` Restart only when the container exit code is non-zero

    """

    _ = ''
    always = 'always'
    unless_stopped = 'unless-stopped'
    on_failure = 'on-failure'
```

**Describe alternatives you've considered**

If we are targeting python 3.8 and above, this issue can be easily mitigated by using `--enum-field-as-literal all`.
Therefore this issue can be resolved by allowing `--enum-field-as-literal` option in python 3.7 by importing `Literal` from typing_extensions.

**Comment by koxudaxi** [comment]
@yuyupopo 
Thank you for creating this issue.
I want to fix it.
I think If the enum value is `''` then, the field name  `empty` is better than `_`.

**Comment by yuyupopo** [comment]
> @yuyupopo
> Thank you for creating this issue.
> I want to fix it.
> I think If the enum value is `''` then, the field name `empty` is better than `_`.

I believe it is highly unlikely, but what if enum is given as below?

```
enum:
          - "" 
          - empty
```

Also, I've tried a quick fix importing `Literal` from `typing_extensions` if target python version is lower than 3.8. 
https://github.com/yuyupopo/datamodel-code-generator feel free to check it out.

**Comment by koxudaxi** [comment]
I just want to say the same problem.

> Also, I've tried a quick fix importing Literal from typing_extensions if target python version is lower than 3.8

I feel a good workaround, But users may need `enum`.

**Comment by yuyupopo** [comment]
Then what about, `--empty-enum-field-value` option that can accept an arbitrary string? The option has `empty` as default (or `_`), and users can change it if they want to.

**Comment by koxudaxi** [comment]
`--empty-enum-field-name` is more strict ?

**Comment by yuyupopo** [comment]
> `--empty-enum-field-name` is more strict ?

Yeah, I believe that's a better name

**Comment by koxudaxi** [comment]
OK, I will fix it soon. Thank you

**Comment by yuyupopo** [comment]
Also, what do you think about this?

> Importing Literal from typing_extensions if target python version is lower than 3.8

Do you feel it necessary? If it does, should I open a new issue, or just create a new PR for it?

**Comment by koxudaxi** [comment]
> Do you feel it necessary? If it does, should I open a new issue, or just create a new PR for it?

I welcome a new issue and PR!!

**Comment by koxudaxi** [comment]
@yuyupopo 
I have released a new version as `0.9.4`
Thank you very much!!

**Comment by yuyupopo** [comment]
Thank you too!
